tests/test_modes.py

```diff
@@ -83,20 +83,37 @@ def test_should_return_command_that_list_all_changed_files(self):
         command = mode.command()
 
         assert isinstance(command, list)
-        assert mode.command() == ["git", "diff", "--name-only", "--relative", "master"]
+        assert mode.command() == [
+            "git",
+            "diff",
+            "--name-status",
+            "--relative",
+            "master",
+        ]
 
-    def test_parser_should_return_the_candidate_itself(self):
+    @pytest.mark.parametrize(
+        "line,expected_line",
+        [
+            ("D       tests/migrations/auto.py", None),
+            (
+                "R098    tests/test_pytest_picked.py     tests/test_pytest_picked.py",
+                "tests/test_pytest_picked.py",
+            ),
+            ("M       test.py", "test.py"),
+            ("AD      test.py", None),
+        ],
+    )
+    def test_parser_should_ignore_no_paths_characters(self, line, expected_line):
         mode = Branch([])
-        line = "tests/test_pytest_picked.py\n"
         parsed_line = mode.parser(line)
 
-        assert parsed_line == line
+        assert parsed_line == expected_line
 
     def test_should_list_branch_changed_files_as_affected_tests(self):
         raw_output = (
-            b"pytest_picked.py\n"
-            + b"tests/test_pytest_picked.py\n"
-            + b"tests/test_other_module.py"
+            b"D       pytest_picked.py\n"
+            + b"R066    tests/test_pytest_picked.py     tests/test_pytest_picked.py\n"
```

Comment:
It would be nice to have different names for the old and the new file name, this way we can make sure that the right file was picked.

```suggestion
            + b"R066    tests/test_pytest_picked.py     tests/test_new_pytest_picked.py\n"
```

Reply:
This is a good idea, thanks!